airbyte-cdk/python/airbyte_cdk/sources/streams/core.py

```diff
@@ -123,6 +124,57 @@ def read_full_refresh(
                 cursor_field=cursor_field,
             )
 
+    def read_incremental(  # type: ignore  # ignoring typing for ConnectorStateManager because of circular dependencies
```

Comment:
This is caused by `connector_state_manager.py` having the statement `from airbyte_cdk.sources.streams import Stream` while `core.py` (where `Stream` is defined) having the statement `from airbyte_cdk.sources.connector_state_manager import ConnectorStateManager`. To be honest, I'm not sure what we can do about this. This comes from the fact that the implementation is leaking in the interface.